kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/api/AdapterGenerator.kt

```diff
@@ -162,6 +163,22 @@ internal class AdapterGenerator(
 
     if (typeVariables.isNotEmpty()) {
       result.addTypeVariables(typeVariables.map { it.stripTypeVarVariance() as TypeVariableName })
+      // require(types.size == 1) {
+      //   "TypeVariable mismatch: Expecting 1 type(s) for generic type variables [T], but received ${types.size} with values $types"
+      // }
+      result.addInitializerBlock(CodeBlock.builder()
+          .beginControlFlow("require(types.size == %L)", typeVariables.size)
+          .addStatement(
+              "buildString·{·append(%S).append(%L).append(%S).append(%S).append(%S).append(%L)·}",
```

Comment:
Won't the generated code look like this?
```kotlin
require(types.size == 1) {
    buildString { append("TypeVariable mismatch: Expecting ").append(1).append(" type for generic type variables [").append("T").append("], but received ").append(typesParamName.size) }
}
```
A bit concerned about this being a very long single line.

Reply:
Yeah, but in the printed error message it looks fine.